core/beacon/types.go

```diff
@@ -183,7 +185,26 @@ func ExecutableDataToBlock(params ExecutableDataV1) (*types.Block, error) {
 
 // BlockToExecutableData constructs the executableDataV1 structure by filling the
 // fields from the given block. It assumes the given block is post-merge block.
-func BlockToExecutableData(block *types.Block) *ExecutableDataV1 {
+func BlockToExecutableData(block *types.Block, fees *big.Int) *ExecutableDataV1 {
+	if fees != nil {
+		return &ExecutableDataV1{
+			BlockHash:     block.Hash(),
+			ParentHash:    block.ParentHash(),
+			FeeRecipient:  block.Coinbase(),
+			StateRoot:     block.Root(),
+			Number:        block.NumberU64(),
+			GasLimit:      block.GasLimit(),
+			GasUsed:       block.GasUsed(),
+			BaseFeePerGas: block.BaseFee(),
+			Timestamp:     block.Time(),
+			ReceiptsRoot:  block.ReceiptHash(),
+			LogsBloom:     block.Bloom().Bytes(),
+			Transactions:  encodeTransactions(block.Transactions()),
+			Random:        block.MixDigest(),
+			ExtraData:     block.Extra(),
+			Fees:          fees,
+		}
```

Comment:
maybe a dumb question, but how is this block different from adding `Fees: fees,` below?

nil is the default value for `Fees`

Reply:
I know, I'm stupid, wrote this after coming home from bogota in the middle of the night^^